### PR TITLE
feat: add lineage config dialog

### PR DESCRIPTION
**Pull request by karanh37**
### Describe your changes :
Closes: #6946 

I worked on the ..... because ...
Added a Config modal to take upstream and downstream depth along with nodes per layer.

#
### Type of change :

- [x] Improvement
- [x] New feature

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers

Frontend: @open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx [hunk 1]
```diff
@@ -146,124 +151,145 @@ const CustomControls: FC<ControlProps> = ({
     );
   }, [isEditMode]);
 
+  const handleDialogSave = useCallback((config: LineageConfig) => {
+    onLineageConfigUpdate(config);
+    setDialogVisible(false);
+  }, []);
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/CustomControls.component.tsx:157]:
```suggestion
  }, [onLineageConfigUpdate, setDialogVisible]);
```

**Review by chirag-madlani — COMMENTED** [review]
Add unit tests for changes

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/NodeSuggestions.component.tsx [hunk 2]
```diff
@@ -50,16 +50,16 @@ const NodeSuggestions: FC<EntitySuggestionProps> = ({
...
-  const getSuggestionLabel = (fqn: string, type: string, name: string) => {
+  const getSuggestionLabelHeading = (fqn: string, type: string) => {
     if (type === EntityType.TABLE) {
       const database = getPartialNameFromTableFQN(fqn, [FqnPart.Database]);
       const schema = getPartialNameFromTableFQN(fqn, [FqnPart.Schema]);
 
       return database && schema
-        ? `${database}${FQN_SEPARATOR_CHAR}${schema}${FQN_SEPARATOR_CHAR}${name}`
-        : name;
+        ? `${database}${FQN_SEPARATOR_CHAR}${schema}`
```
Comment (chirag-madlani) [inline openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/NodeSuggestions.component.tsx:59]: Need to verify this, We will only show it for database & schema

**Review by chirag-madlani — COMMENTED** [review]
@karanh37 add screenshots for UI changes. Left one minor comment will get confirmation

**Comment by karanh37** [comment]
UI changes screenshots

- Node style has been changed. Database and Schema are showed above the name.
- Lineage config modal has been added.
- Styles for dropdown has been changed.

Screenshot 2023-03-16 at 11 04 51 AM
Screenshot 2023-03-16 at 11 05 17 AM
Screenshot 2023-03-16 at 11 05 44 AM
Screenshot 2023-03-16 at 11 05 54 AM
Screenshot 2023-03-16 at 11 06 01 AM
Screenshot 2023-03-16 at 11 06 08 AM
Screenshot 2023-03-16 at 11 06 25 AM

File: openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/NodeSuggestions.component.tsx (hunk 2 above)
Comment (karanh37) [reply openmetadata-ui/src/main/resources/ui/src/components/EntityLineage/NodeSuggestions.component.tsx:59]: Verified on existing implementation. We only show for database & schema.

**Review by chirag-madlani — APPROVED** [review]

**Comment by ShaileshParmar11** [comment]
> UI changes screenshots
> 
> * Node style has been changed. Database and Schema are showed above the name.
> * Lineage config modal has been added.
> * Styles for dropdown has been changed.
> 
> Screenshot 2023-03-16 at 11 04 51 AM Screenshot 2023-03-16 at 11 05 17 AM Screenshot 2023-03-16 at 11 05 44 AM Screenshot 2023-03-16 at 11 05 54 AM Screenshot 2023-03-16 at 11 06 01 AM Screenshot 2023-03-16 at 11 06 08 AM Screenshot 2023-03-16 at 11 06 25 AM

Hi @karanh37, can you check why its showing default icon for mlmodal and topic entity

**Review by ShaileshParmar11 — CHANGES_REQUESTED** [review]

**Pull request merged**